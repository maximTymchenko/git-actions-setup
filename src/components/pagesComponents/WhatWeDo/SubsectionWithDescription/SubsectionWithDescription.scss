@import '../../../../styles/vars';

.subsection-with-description {
    @include section__size;
    margin-top: 60px;
    margin-bottom: 60px;
    position: relative;

    &:first-child {
        margin-right: 100px;
    }

    &::before {
        position: absolute;
        top: -30px;
        left: -20px;
        content: url('../../../../assets/img/subsectionWithDescription/subsection-waves-bg.svg');
    }
}
