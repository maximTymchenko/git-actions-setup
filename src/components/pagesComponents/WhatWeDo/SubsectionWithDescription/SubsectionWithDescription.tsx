import React, { FC } from 'react';
import SectionTitle from '../../../common/SectionTitle';
import TextDescription from '../../../common/TextDescription';
import './SubsectionWithDescription.scss';

interface ISubsectionWithDescription {
    title: string;
    description: string;
}

const SubsectionWithDescription: FC = (props: ISubsectionWithDescription) => {
    return (
        <div data-testid="subsection-with-description" className="subsection-with-description">
            <SectionTitle type="primary" title={props.title} />
            <TextDescription description={props.description} />
        </div>
    );
};

export default SubsectionWithDescription;
