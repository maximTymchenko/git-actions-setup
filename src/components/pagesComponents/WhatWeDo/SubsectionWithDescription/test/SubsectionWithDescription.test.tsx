import React from 'react'
import SubsectionWithDescription from '../SubsectionWithDescription'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
it('renders elem correctly', () => {
  const { getByTestId } = render(
    <SubsectionWithDescription
      title="Retail and E-commerce"
      description="Our clients’ products are heavily used by such companies like Walmart, The Home Depot, CVS Pharmacy in the USA and Canada. It’s a joy to be a part of their success."
    />
  )
  expect(getByTestId('subsection-with-description')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer
    .create(
      <SubsectionWithDescription
        title="Retail and E-commerce"
        description="Our clients’ products are heavily used by such companies like Walmart, The Home Depot, CVS Pharmacy in the USA and Canada. It’s a joy to be a part of their success."
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
