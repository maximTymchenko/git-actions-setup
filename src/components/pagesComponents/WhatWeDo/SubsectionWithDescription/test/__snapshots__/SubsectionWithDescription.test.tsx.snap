// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="subsection-with-description"
  data-testid="subsection-with-description"
>
  <h2
    className="title title--primary"
    data-testid="section-title"
  >
    Retail and E-commerce
  </h2>
  <p
    className="text-description text-description--undefined text-description--undefined"
    data-testid="text-description"
  />
</div>
`;
